src/Billing/EDI_270.php

```diff
@@ -0,0 +1,1144 @@
+<?php
+/**
+ * Inc file for the 270 / 271 creation and uploading
+ *
+ * This program creates the segments for the x12 270 eligibility file
+ * It also allows the reading and storing of the x12 271 file
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Terry Hill <terry@lilysystems.com>
+ * @author    Brady Miller <brady.g.miller@gmail.com>
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @author    Stephen Waite <stephen.waite@cmsvt.com>
+ * @copyright Copyright (c) 2010 MMF Systems, Inc
+ * @copyright Copyright (c) 2016 Terry Hill <terry@lillysystems.com>
+ * @copyright Copyright (c) 2017 Brady Miller <brady.g.miller@gmail.com>
+ * @copyright Copyright (c) 2019 Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2019 Stephen Waite <stephen.waite@cmsvt.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+namespace OpenEMR\Billing;
+
+require_once(dirname(__FILE__) . "/../../library/edihistory/codes/edih_271_code_class.php");
+
+use OpenEMR\Common\Http\oeHttp;
+use OpenEMR\Common\Utils\RandomGenUtils;
+
+// @TODO global to become private var when this goes to a class.
+//
+class EDI_270
```

Comment:
Although these new classes don't rise to an abstract yet, they sure act as singletons(basically a static class here). As such, I recm that they are prevented from being instantiated.
Could add a private function __construct {} .

class EDI_270
{
    private function __construct() {}

May be overkill but, in reality by replacing script includes with a class then i'd think the class needs to be a singleton  where you'd grab a instance of the class and address static class methods via the instance reference. This way, the class and any variables referenced will be a known state with only one instance. An example of this can be seen with _rest_config.php.
Think in terms of if you require() a script opposed to require_once(). Not sure if this use case will run into issues but I thought i'd point it out. I still think in c++:)

Reply:
ok, thanks guys